cmd/main.go

```diff
@@ -103,28 +114,52 @@ func main() {
 	}
 	setupLog.Info("parsed kubernetes templates", "templates", template.DefinedTemplates())
 
-	discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(mgr.GetConfig())
+	mgrRestConfig := mgr.GetConfig()
+	client := mgr.GetClient()
+	tokenReview := &v1.TokenReview{
+		Spec: v1.TokenReviewSpec{
+			Token: mgrRestConfig.BearerToken,
+		},
+	}
+	err = client.Create(context.Background(), tokenReview)
+	if err != nil {
+		setupLog.Error(err, "error getting controller serviceaccount audience")
+		os.Exit(1)
+	}
+	setupLog.Info("default authorino authconfig audiences", "audiences", tokenReview.Status.Audiences)
+
+	discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(mgrRestConfig)
 	groups, err := discoveryClient.ServerGroups()
 	if err != nil {
 		setupLog.Error(err, "error discovering server groups")
 		os.Exit(1)
 	}
 	isOpenShift := false
+	hasAuthorino := false
+	hasIstio := false
 	for _, g := range groups.Groups {
 		if g.Name == "route.openshift.io" {
 			isOpenShift = true
 		}
+		if g.Name == "authorino.kuadrant.io" {
+			hasAuthorino = true
+		}
+		if g.Name == "networking.istio.io" {
+			hasIstio = true
+		}
```

Comment:
but isnt this only saying API server has this been registered 
 it could be a case, "authorino operator was installed and then uninstalled" still marks it to be true ?

Reply:
Good point @zdtsw . This check is used to create watches in `SetupWithManager()` in main. If authorino/istio are uninstalled, the operator is going to go into a loop trying to reconcile model registries that use istio config. 
It would be easier to restart the operator, than to try and make the operator dynamically look for Authorino/Istio install/uninstall events and add/remove watches in the controller manager. 
Since opendatahub operator handles mesh install/uninstall, perhaps we can add that logic in there. Maybe in a featuretracker we can restart the model registry operator. wdyt?

Reply:
sure, i think if MR is deployed as part of ODH, ODH operator can deal with this kind of checks.
but if you want plan to have Model Registry operator to be more independent, you will still need some logic for the check. 
but for the former case, it can follow the same patter as other components.

Reply:
Created issue https://issues.redhat.com/browse/RHOAIENG-5110 to track this edge case of Istio/Authorino uninstall.